code/processes/segmentedtickerplant.q

```diff
@@ -51,10 +67,11 @@ subdetails:{[tabs;instruments]
 
 init:{[b]
   if[not all b in/:(key .stplg.upd;key .stplg.zts);'"mode ",(string b)," must be defined in both .stplg.upd and .stplg.zts"];
+  .u.init[.stpps.t];
   .stplg.updmsg:.stplg.upd[b];
   .u.upd:{[t;x]
     // snap the current time and check for period end
-    if[.stplg.nextendUTC<now:.z.p;.stplg.checkends now];
+    if[.stplg.nextendUTC<now:.z.p and not chainedtp;.stplg.checkends now];
```

Comment:
Change this to have a switch for the defintion of .u.upd depending on whether we are in SCTP or STP mode